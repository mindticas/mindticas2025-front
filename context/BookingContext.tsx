import React, { createContext, useState, useContext } from 'react';

// Define el tipo de datos que se almacenarÃ¡n en el contexto
type BookingContextType = {
    service: string;
    setService: (service: string) => void;
    dateTime: Date | null;
    setDateTime: (dateTime: Date | null) => void;
    personData: { name: string; lastName: string; phone: string };
    setPersonData: (personData: {
        name: string;
        lastName: string;
        phone: string;
    }) => void;
};

// Crea el contexto
const BookingContext = createContext<BookingContextType | undefined>(undefined);

// Hook personalizado para usar el contexto
export const useBookingContext = () => {
    const context = useContext(BookingContext);
    if (!context) {
        throw new Error(
            'useBookingContext must be used within a BookingProvider',
        );
    }
    return context;
};

// Proveedor del contexto
export const BookingProvider: React.FC<{ children: React.ReactNode }> = ({
    children,
}) => {
    const [service, setService] = useState<string>('');
    const [dateTime, setDateTime] = useState<Date | null>(null);
    const [personData, setPersonData] = useState<{
        name: string;
        lastName: string;
        phone: string;
    }>({
        name: '',
        lastName: '',
        phone: '',
    });

    return (
        <BookingContext.Provider
            value={{
                service,
                setService,
                dateTime,
                setDateTime,
                personData,
                setPersonData,
            }}
        >
            {children}
        </BookingContext.Provider>
    );
};
