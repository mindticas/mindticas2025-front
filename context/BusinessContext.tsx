'use client';

import { UserProfile } from '@/interfaces/userProfile/userProfile';
import { getUserProfile } from '@/services/userProfileService';
import React, {
    createContext,
    useContext,
    useState,
    ReactNode,
    useEffect,
} from 'react';

interface BusinessInfo {
    name: string;
    address: string;
    phone: string;
    instagram?: string;
}

interface BusinessContextType {
    businessInfo: BusinessInfo;
    setBusinessInfo: (info: BusinessInfo) => void;
    isLoading: boolean;
}

export const BusinessContext = createContext<BusinessContextType | undefined>(
    undefined,
);

export const useBusiness = () => {
    const context = useContext(BusinessContext);
    if (!context) {
        throw new Error('useBusiness must be used within a BusinessProvider');
    }
    return context;
};

export const BusinessProvider: React.FC<{ children: ReactNode }> = ({
    children,
}) => {
    const [businessInfo, setBusinessInfo] = useState<BusinessInfo>({
        name: '',
        address: '',
        phone: '',
        instagram: '',
    });
    const [isLoading, setIsLoading] = useState<boolean>(false);
    const fetchUserProfile = async () => {
        setIsLoading(true);

        try {
            const data = await getUserProfile();
            if (!Array.isArray(data) || data.length === 0) {
                console.error('No se encontrÃ³ perfil del negocio');
                return;
            }
            const profile: UserProfile = data[0];
            setBusinessInfo({
                name: profile.name ?? '',
                address: profile.contactDetails?.address ?? '',
                phone: profile.contactDetails?.phone ?? '',
                instagram: profile.socialLinks?.instagram ?? '',
            });
        } catch (error) {
            console.error('Error al cargar el perfil del usuario', error);
        } finally {
            setIsLoading(false);
        }
    };

    useEffect(() => {
        fetchUserProfile();
    }, []);

    return (
        <BusinessContext.Provider
            value={{ businessInfo, setBusinessInfo, isLoading }}
        >
            {children}
        </BusinessContext.Provider>
    );
};
export type { BusinessInfo };
