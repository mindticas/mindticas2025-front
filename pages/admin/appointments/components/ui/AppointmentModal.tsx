'use client';

import { useEffect, useState } from 'react';
import {
    Dialog,
    Portal,
    Box,
    Input,
    Button,
    Text,
    Field,
    Spinner,
    SelectRoot,
    SelectContent,
    SelectItem,
    SelectTrigger,
    SelectLabel,
    CloseButton,
} from '@chakra-ui/react';
import { Treatment } from '@/interfaces/treatment/Treatment';
import { createListCollection } from '@chakra-ui/react';

interface AppointmentModalProps {
    isOpen: boolean;
    onClose: () => void;
    mode: 'create' | 'edit' | 'cancel';
    treatments: Treatment[];
    initialData?: {
        id?: number;
        name: string;
        treatment: string;
        date: string;
        time: string;
    };
    onConfirm: (data: {
        id?: number;
        name: string;
        treatment: string;
        date: string;
        time: string;
    }) => Promise<void>;
    onCancel?: (id: number) => Promise<void>;
}

export default function AppointmentModal({
    isOpen,
    onClose,
    mode,
    treatments,
    initialData = {
        name: '',
        treatment: '',
        date: '',
        time: '',
    },
    onConfirm,
    onCancel,
}: AppointmentModalProps) {
    const [formData, setFormData] = useState({
        name: initialData.name,
        treatment: initialData.treatment,
        date: initialData.date,
        time: initialData.time,
    });

    useEffect(() => {
        if (isOpen) {
            setFormData({
                name: initialData.name,
                treatment: initialData.treatment,
                date: initialData.date,
                time: initialData.time,
            });
        }
    }, [
        isOpen,
        initialData.id,
        initialData.name,
        initialData.treatment,
        initialData.date,
        initialData.time,
    ]);

    const [isLoading, setIsLoading] = useState(false);

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setFormData((prev) => ({ ...prev, [name]: value }));
    };

    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault();
        setIsLoading(true);
        try {
            await onConfirm({
                id: initialData?.id,
                ...formData,
            });
            onClose();
        } finally {
            setIsLoading(false);
        }
    };

    const handleCancelAppointment = async () => {
        if (mode === 'cancel' && initialData?.id && onCancel) {
            setIsLoading(true);
            try {
                await onCancel(initialData.id);
                onClose();
            } finally {
                setIsLoading(false);
            }
        }
    };

    return (
        <Dialog.Root
            open={isOpen}
            onOpenChange={({ open }) => !open && onClose()}
            placement='center'
        >
            <Portal>
                <Dialog.Backdrop />
                <Dialog.Positioner>
                    <Dialog.Content p={5} backgroundColor='white'>
                        <Dialog.Header
                            mb={4}
                            display='flex'
                            justifyContent='space-between'
                            alignContent='center'
                        >
                            <Dialog.Title>
                                {mode === 'cancel'
                                    ? 'Confirmar cancelaci√≥n'
                                    : mode === 'edit'
                                    ? 'Editar Cita'
                                    : 'Nueva Cita'}
                            </Dialog.Title>
                            <Dialog.CloseTrigger asChild>
                                <CloseButton color='black' size='sm' />
                            </Dialog.CloseTrigger>
                        </Dialog.Header>

                        <Dialog.Body>
                            {mode !== 'cancel' ? (
                                <form onSubmit={handleSubmit}>
                                    <Box mb={4}>
                                        <Field.Root>
                                            <Field.Label fontWeight='semibold'>
                                                Nombre
                                            </Field.Label>
                                            <Input
                                                p={2}
                                                name='name'
                                                placeholder='Nombre del Cliente'
                                                value={formData.name}
                                                onChange={handleInputChange}
                                                required
                                            />
                                        </Field.Root>
                                    </Box>

                                    <Box mb={4}>
                                        {treatments && (
                                            <SelectRoot
                                                name='treatment'
                                                collection={createListCollection(
                                                    {
                                                        items: treatments,
                                                    },
                                                )}
                                                onValueChange={(e) => {
                                                    const selectedTreatment =
                                                        treatments.find(
                                                            (treatment) =>
                                                                treatment.name ===
                                                                e.value.toString(),
                                                        );
                                                    setFormData({
                                                        ...formData,
                                                        treatment:
                                                            selectedTreatment
                                                                ? selectedTreatment.id.toString()
                                                                : '',
                                                    });
                                                }}
                                                required
                                            >
                                                <SelectLabel fontWeight='semibold'>
                                                    Servicio a realizar
                                                </SelectLabel>
                                                <SelectTrigger>
                                                    {(treatments &&
                                                        treatments.find(
                                                            (t) =>
                                                                t.id.toString() ===
                                                                formData.treatment,
                                                        )?.name) ||
                                                        'Seleccionar servicio'}
                                                </SelectTrigger>
                                                <SelectContent
                                                    className='select-content-admin-edit'
                                                    backgroundColor='white'
                                                >
                                                    {treatments &&
                                                        treatments.map(
                                                            ({ id, name }) => (
                                                                <SelectItem
                                                                    cursor='pointer'
                                                                    _hover={{
                                                                        backgroundColor:
                                                                            'gray.100',
                                                                    }}
                                                                    backgroundColor='white'
                                                                    item={name}
                                                                    key={id}
                                                                    p={2}
                                                                    data-state={
                                                                        id ===
                                                                        id
                                                                            ? 'checked'
                                                                            : 'unchecked'
                                                                    }
                                                                >
                                                                    {name}
                                                                </SelectItem>
                                                            ),
                                                        )}
                                                </SelectContent>
                                            </SelectRoot>
                                        )}
                                    </Box>

                                    <Box mb={4}>
                                        <Field.Root>
                                            <Field.Label fontWeight='semibold'>
                                                Fecha
                                            </Field.Label>
                                            <Input
                                                _dark={{
                                                    '&::-webkit-calendar-picker-indicator':
                                                        {
                                                            filter: 'invert(1)',
                                                        },
                                                }}
                                                p={2}
                                                type='date'
                                                name='date'
                                                value={formData.date}
                                                onChange={handleInputChange}
                                                required
                                            />
                                        </Field.Root>
                                    </Box>

                                    <Box mb={4}>
                                        <Field.Root>
                                            <Field.Label fontWeight='semibold'>
                                                Hora
                                            </Field.Label>
                                            <Input
                                                _dark={{
                                                    '&::-webkit-calendar-picker-indicator':
                                                        {
                                                            filter: 'invert(1)',
                                                        },
                                                }}
                                                p={2}
                                                type='time'
                                                name='time'
                                                value={formData.time}
                                                onChange={handleInputChange}
                                                required
                                            />
                                        </Field.Root>
                                    </Box>
                                </form>
                            ) : (
                                <Text>
                                    ¬øEst√°s seguro que deseas cancelar esta cita?
                                </Text>
                            )}
                        </Dialog.Body>

                        <Dialog.Footer p={4}>
                            <Button
                                backgroundColor={
                                    mode === 'cancel' ? 'black' : 'red'
                                }
                                color='white'
                                p={3}
                                onClick={onClose}
                            >
                                {mode === 'cancel' ? 'Volver' : 'Cancelar'}
                            </Button>

                            <Button
                                colorScheme={mode === 'cancel' ? 'red' : 'blue'}
                                ml={3}
                                onClick={
                                    mode === 'cancel'
                                        ? handleCancelAppointment
                                        : handleSubmit
                                }
                                p={3}
                                backgroundColor={
                                    mode === 'cancel' ? 'red' : '#1C4ED8'
                                }
                                color='white'
                                disabled={isLoading}
                            >
                                {isLoading ? (
                                    <Spinner size='sm' mr={2} />
                                ) : mode === 'cancel' ? (
                                    'Confirmar cancelaci√≥n'
                                ) : mode === 'edit' ? (
                                    'Actualizar'
                                ) : (
                                    'Guardar'
                                )}
                            </Button>
                        </Dialog.Footer>
                    </Dialog.Content>
                </Dialog.Positioner>
            </Portal>
        </Dialog.Root>
    );
}
