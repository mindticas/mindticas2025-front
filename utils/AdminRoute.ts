import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
