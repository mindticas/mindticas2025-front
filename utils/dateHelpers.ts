// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}