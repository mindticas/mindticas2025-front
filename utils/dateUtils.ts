import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
