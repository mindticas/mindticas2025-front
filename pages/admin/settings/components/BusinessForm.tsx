import {
    Box,
    VStack,
    Text,
    Flex,
    Button,
    Spinner,
    Input,
    Center,
    Skeleton,
} from '@chakra-ui/react';
import React, { useEffect, useState } from 'react';
import { updateUserProfile } from '@/services/userProfileService';
import {
    BusinessFormData,
    hasBusinessFormChanges,
    showBusinessNotification,
    validateBusinessFields,
} from '@/utils/userProfile/userProfileValidation';
import { BusinessInfo } from '@/context/BusinessContext';

interface BusinessFormProps {
    businessInfo: BusinessInfo;
    setBusinessInfo: (info: BusinessInfo) => void;
    isSmallScreen: boolean;
    isLoading: boolean;
}

export default function BusinessForm({
    businessInfo,
    setBusinessInfo,
    isSmallScreen,
    isLoading,
}: BusinessFormProps) {
    const [name, setName] = useState('');
    const [address, setAddress] = useState('');
    const [phone, setPhone] = useState('');
    const [instagram, setInstagram] = useState('');
    const [isSaving, setIsSaving] = useState(false);

    useEffect(() => {
        setName(businessInfo.name);
        setAddress(businessInfo.address);
        setPhone(businessInfo.phone);
        setInstagram(businessInfo.instagram || '');
    }, [businessInfo]);

    const handleSave = async () => {
        setIsSaving(true);
        const trimmedData: BusinessFormData = {
            name: name.trim(),
            contactDetails: {
                address: address.trim(),
                phone: phone.trim(),
            },
            socialLinks: {
                instagram: instagram.trim(),
            },
        };
        // Validation: empty fields
        const validation = validateBusinessFields(trimmedData);
        if (!validation.isValid) {
            setIsSaving(false);
            return showBusinessNotification.validationError(validation.error!);
        }
        // Check for changes in inputs
        if (!hasBusinessFormChanges(businessInfo, trimmedData)) {
            setIsSaving(false);
            return showBusinessNotification.noChanges();
        }
        // If validations pass, save
        try {
            const updated = await updateUserProfile(trimmedData);
            setBusinessInfo({
                name: updated.name || '',
                address: updated.contactDetails?.address || '',
                phone: updated.contactDetails?.phone || '',
                instagram: updated.socialLinks?.instagram || '',
            });
            showBusinessNotification.saveSuccess();
        } catch (error) {
            console.log(error);
            showBusinessNotification.saveError();
        } finally {
            setIsSaving(false);
        }
    };

    return (
        <Box
            width={isSmallScreen ? '100%' : '70%'}
            p={6}
            bg='white'
            borderRadius='lg'
            borderWidth='1px'
            borderColor='gray.200'
            shadow='md'
        >
            {isLoading ? (
                <VStack gap={6} align='stretch'>
                    <Skeleton height='50px' />
                    <Skeleton height='50px' />
                    <Skeleton height='50px' />
                    <Skeleton height='50px' />
                    <Center mt={4}>
                        <Spinner size='xl' color='#1C4ED8' />
                    </Center>
                </VStack>
            ) : (
                <VStack gap={4} align='stretch'>
                    <Box>
                        <Text fontWeight='semibold' mb={2} color='gray.700'>
                            Nombre del negocio
                        </Text>
                        <Input
                            value={name}
                            onChange={(e) => setName(e.target.value)}
                            p={3}
                            borderRadius='md'
                            borderColor='gray.300'
                        />
                    </Box>

                    <Box>
                        <Text fontWeight='semibold' mb={2} color='gray.700'>
                            Dirección
                        </Text>
                        <Input
                            value={address}
                            onChange={(e) => setAddress(e.target.value)}
                            p={3}
                            borderRadius='md'
                            borderColor='gray.300'
                        />
                    </Box>

                    <Box>
                        <Text fontWeight='semibold' mb={2} color='gray.700'>
                            Teléfono
                        </Text>
                        <Input
                            value={phone}
                            onChange={(e) => setPhone(e.target.value)}
                            type='number'
                            p={3}
                            borderRadius='md'
                            borderColor='gray.300'
                        />
                    </Box>

                    <Box>
                        <Text fontWeight='semibold' mb={2} color='gray.700'>
                            Instagram
                        </Text>
                        <Input
                            value={instagram}
                            onChange={(e) => setInstagram(e.target.value)}
                            p={3}
                            borderRadius='md'
                            borderColor='gray.300'
                        />
                    </Box>

                    <Flex justify='center' mt={6}>
                        <Button
                            bgColor='#1C4ED8'
                            color='white'
                            size='md'
                            px={8}
                            py={4}
                            fontWeight='medium'
                            borderRadius='md'
                            onClick={handleSave}
                            _hover={{
                                transform: 'translateY(-2px)',
                                boxShadow: 'md',
                            }}
                            transition='all 0.2s'
                            loading={isSaving}
                        >
                            {isSaving ? (
                                <Flex align='center'>
                                    <Spinner size='sm' color='white' mr={2} />
                                    <Text>Guardando...</Text>
                                </Flex>
                            ) : (
                                'Guardar cambios'
                            )}
                        </Button>
                    </Flex>
                </VStack>
            )}
        </Box>
    );
}
