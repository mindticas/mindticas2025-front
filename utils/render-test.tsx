import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
