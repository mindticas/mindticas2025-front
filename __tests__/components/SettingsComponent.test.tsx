import React from 'react';
import { screen } from '@testing-library/react';
import Settings from '@/pages/admin/settings';
import { useBusiness } from '@/context/BusinessContext';
import { Toaster } from '@/components/ui/toaster';
import BusinessForm from '@/pages/admin/settings/components/BusinessForm';
import { render } from '@/utils/render-test';

jest.mock('@/context/BusinessContext');
jest.mock('@/components/ui/toaster');
jest.mock('@/pages/admin/settings/components/BusinessForm');
jest.mock('@chakra-ui/react', () => ({
    ...jest.requireActual('@chakra-ui/react'),
    useBreakpointValue: jest.fn(),
}));

const mockUseBusiness = useBusiness as jest.MockedFunction<typeof useBusiness>;
const mockUseBreakpointValue =
    jest.requireMock('@chakra-ui/react').useBreakpointValue;

describe('Settings Page', () => {
    const mockBusinessInfo = {
        name: 'Mi Negocio',
        address: 'Calle Principal 123',
        phone: '123456789',
        instagram: '@minegocio',
    };

    beforeEach(() => {
        // Common mock configuration
        mockUseBusiness.mockReturnValue({
            businessInfo: mockBusinessInfo,
            setBusinessInfo: jest.fn(),
            isLoading: false,
        });

        mockUseBreakpointValue.mockImplementation(() => false);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render correctly on large screens', () => {
        const { asFragment } = render(<Settings />);

        expect(asFragment()).toMatchSnapshot();
        expect(screen.getByText('InformaciÃ³n del negocio')).toBeInTheDocument();
        expect(BusinessForm).toHaveBeenCalledWith(
            expect.objectContaining({
                businessInfo: mockBusinessInfo,
                isSmallScreen: false,
                isLoading: false,
            }),
            expect.anything(),
        );
    });

    it('should render correctly on mobile', () => {
        mockUseBreakpointValue.mockImplementation(() => true);
        const { asFragment } = render(<Settings />);

        expect(asFragment()).toMatchSnapshot();
        expect(BusinessForm).toHaveBeenCalledWith(
            expect.objectContaining({
                isSmallScreen: true,
            }),
            expect.anything(),
        );
    });

    it('should show the charging status', () => {
        mockUseBusiness.mockReturnValue({
            businessInfo: mockBusinessInfo,
            setBusinessInfo: jest.fn(),
            isLoading: true,
        });

        render(<Settings />);
        expect(BusinessForm).toHaveBeenCalledWith(
            expect.objectContaining({
                isLoading: true,
            }),
            expect.anything(),
        );
    });

    it('must include the Toaster component', () => {
        render(<Settings />);
        expect(Toaster).toHaveBeenCalled();
    });
});
