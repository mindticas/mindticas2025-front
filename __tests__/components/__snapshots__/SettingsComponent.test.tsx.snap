// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page debe renderizar correctamente en mobile 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="light css-1i6ke5x"
  >
    <h2
      class="chakra-heading css-1xo4whg"
    >
      Información del negocio
    </h2>
    <div
      class="css-dy8jbg"
    />
  </div>
</DocumentFragment>
`;

exports[`Settings Page debe renderizar correctamente en pantallas grandes 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="light css-laq3g5"
  >
    <h2
      class="chakra-heading css-1xo4whg"
    >
      Información del negocio
    </h2>
    <div
      class="css-dy8jbg"
    />
  </div>
</DocumentFragment>
`;
